### PR TITLE
Implemented IsParallelTo, IsPerpendicularTo, IsCollinearTo, IsCoplana…

### PR DESCRIPTION
**Pull request by kabircosta**
…rTo methods and Arc2D class. Tests are still pending.

### CODE REVIEW

**Comment by jkalias** [comment]
Thanks for your effort.

There is no `Line2D` and `Line3D` type anymore, just `Line` (a general 3D line). I am in a trajectory of eliminating all 2D/3D redundancies (according to #200 ) so the master branch is changing quite rapidly lately.

File: src/Spatial/Euclidean/Plane.cs [hunk 1]
```diff
@@ -410,6 +410,66 @@ public Point3D IntersectionWith(Ray3D ray, double tolerance = float.Epsilon)
...
+
+        /// <summary>
+        /// Checks if the plane is coplanar to another plane.
+        /// </summary>
+        /// <param name="other">The plane to test against.</param>
+        /// <param name="angleTolerance">The angle tolerance to use for comparison.</param>
+        /// <param name="tolerance">The tolerance to use for comparison.</param>
+        /// <returns>True if the planes are coplanar, false otherwise.</returns>
+        [Pure]
+        public bool IsCoplanarTo(Plane other, Angle angleTolerance = Precision.DoublePrecision * 2, double tolerance = double.Epsilon)
```
Comment (jkalias) [inline src/Spatial/Euclidean/Plane.cs:467]: This is essentially the implementation of `Equals`, this is not needed

**Review by jkalias — CHANGES_REQUESTED** [review]

File: src/Spatial/Euclidean/Plane.cs (hunk 1 above)
Comment (kabircosta) [reply src/Spatial/Euclidean/Plane.cs:467]: Two planes can be coplanar, but in opposite directions.
For example:
Case 1) Coplanar and same direction: normal1 = normal2
Case 2) Coplanar and opposite direction: normal1 = normal2.Inverse

**Comment by jkalias** [comment]
After the discussion in #229 and the reverted changes, we might have to revisit the comments here

**Comment by jkalias** [comment]
Could you please split this PR into two distinct ones? The one containing only the additional functions, and another containing the new ```Arc2D``` class? It would then be easier to separate the concerns

**Pull request closed without merging**